src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -0,0 +1,126 @@
+            _formatter = formatter;
+        }
+
+        public async Task WriteAsync(IEnumerable<LogEvent> events)
+        {
+            var chunks = SerializeEvents(events);
+            var tasks = chunks.Select(Post);
+
+            await Task.WhenAll(tasks).ConfigureAwait(false);
+        }
+
+        private List<String> SerializeEvents(IEnumerable<LogEvent> events)
+        {
+            List<string> chunks = new List<string>();
+
+            int currentSize = 0;
+
+            var chunkBuffer = new List<string>(events.Count());
+            foreach (var logEvent in events)
+            {
+                var formattedLog = _formatter.formatMessage(logEvent);
+                var logSize = Encoding.UTF8.GetByteCount(formattedLog);
+                if (logSize > _maxMessageSize)
+                {
+                    continue;  // The log is dropped because the backend would not accept it
+                }
+                if (currentSize + logSize > _maxSize)
+                {
+                    // Flush the chunkBuffer to the chunks and reset the chunkBuffer
+                    chunks.Add(GenerateChunk(chunkBuffer, ",", "[", "]"));
+                    chunkBuffer = new List<string>(events.Count());
+                    currentSize = 0;
+                }
+                chunkBuffer.Add(formattedLog);
+                currentSize += logSize;
+            }
+            chunks.Add(GenerateChunk(chunkBuffer, ",", "[", "]"));
+
+            return chunks;
+
+        }
+
+        private static string GenerateChunk(IEnumerable<string> collection, string delimiter, string prefix, string suffix)
+        {
+            return prefix + String.Join(delimiter, collection) + suffix;
+        }
+
+        private async Task Post(string payload)
+        {
+            var content = new StringContent(payload, Encoding.UTF8, _content);
+            for (int retry = 0; retry < MaxRetries; retry++)
+            {
+                int backoff = (int)Math.Min(Math.Pow(retry, 2), MaxBackoff);
+                if (retry > 0)
+                {
+                    await Task.Delay(backoff * 1000);
+                }
+
+                try
+                {
+                    SelfLog.WriteLine("Sending payload to Datadog: {0}", payload);
+                    var result = await _client.PostAsync(_url, content);
+                    SelfLog.WriteLine("Statuscode: {0}", result.StatusCode);
+                    if (result == null) { continue; }
+                    if ((int)result.StatusCode >= 500) { continue; }
+                    if ((int)result.StatusCode >= 400) { break; }
+                    if (result.IsSuccessStatusCode) { return; }
+                }
+                catch (Exception)
+                {
+                    continue;
+                }
+            }
+            SelfLog.WriteLine("Could not send payload to Datadog: {0}", payload);
+        }
+
+        void IDatadogClient.Close() {}
```

Comment:
can you open a session ? are you reusing the same TCP connection (HTTP/1.1) or is there 1 connection per request (HTTP/1.0)?

Reply:
Sessions are handled by the underlying `HttpClient`: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.7.2#remarks